### ISSUE TITLE
Input image n4out not valid

### ISSUE THREAD

**Issue by hajerkr** [mh-cad/vistarsier#24]
I am having this error WRT my input images (I think). I am providing 2 .nii files  but it doesn't seem to be the format it's looking for? Could you help me figure out what the issue may be ? Many thanks.

![image](https://user-images.githubusercontent.com/13291010/124942344-73f6a600-e003-11eb-9e62-6efda76170ab.png)

**Comment by plwp** [comment]
Hi,

I think the most likely mistake would be that you're not mounting the docker volume to your input data. The full docker command should look something like:

`docker run -it --rm -v /full/path/to/your/data:/data plwp/vistarsier /data/prior.nii.gz /data/current.nii.gz /data/`

A couple of other things to note:

- The output prefix will just be added to the filename as-is so if it's a directory you'll need to add a slash at the end i.e `/data/` not `/data`
- It looks like you're using the python version (pytarsier vs. vistarsier). The python version has a weird clipping issue on the output so I wouldn't recommend using it.

**Comment by hajerkr** [comment]
Thanks for the quick response @plwp .

I did mount the docker volume to my input data by running

` singularity shell -B /rds/general/user/hk3618/projects/adniplus/live/MS_NHS/data/FLAIR:/data pytarsier_latest.sif
`

I am running this via singularity so the call is slightly different:

`singularity run pytarsier_latest.sif rds/general/user/hk3618/projects/adniplus/live/MS_NHS/data/FLAIR/Image_2287596084_MR_2453913148_T2W_FLAIR_sag.nii rds/general/user/hk3618/projects/adniplus/live/MS_NHS/data/FLAIR/Image_2287596084_MR_9219981232_T2W_FLAIR.nii /data/`

I tried both relative and full path to point to the data but I still get the same errors. Initially I thought it was because I had compressed nifti files, nii.gz, so I unzipped them and provided nii

- The reason why I ran pytarsier is because the other image did not work for me on Linux, I think it may be a .NET issue from what I googled: 

```
singularity run vistarsier_latest.sif rds/general/user/hk3618/projects/adniplus/live/MS_NHS/data/FLAIR/Image_2287596084_MR_2453913148_T2W_FLAIR_sag.nii rds/general/user/hk3618/projects/adniplus/live/MS_NHS/data/FLAIR/Image_2287596084_MR_9219981232_T2W_FLAIR.nii /data/
Failed to create CoreCLR, HRESULT: 0x80004005
```

I originally pulled it by running ` singularity pull docker://plwp//vistarsier `

**Comment by plwp** [comment]
I haven’t had any experience with singularity so I can’t help much from that end. If any of this helps:

- The vistarsier docker is all Linux-based so that shouldn’t be the issue (I’m running Linux Mint on my main dev machine)
- I was able to recreate the error by not mounting the volume (so no access to the files), but I think the error is coming from the ants n4biasfieldcorrection command.
- I would maybe try to do a test run in docker (if you can) or ants directly just to isolate the problem


Hope any of that is useful

**Comment by hajerkr** [comment]
Hey @plwp , thanks a lot for the feedback. I actually tried running it in docker (on my windows machine), but I get the error reported here https://github.com/mh-cad/vistarsier/issues/15 , where you advised pulling pytarsier. That's why I tried pulling pytarsier on Linux. Essentially, since the prebuilt image didn't work (`singuIarity pull docker://plwp/vistarsier `), I thought about downloading the image via docker on my windows, build it for Linux using Dockerfile, then pull from the HPC using my repo, assuming there is something wrong with the pre-built one..
But I am out of ideas!
